src/coord/rmr/redis_cluster.c

```diff
@@ -190,17 +218,29 @@ static MRClusterTopology *RedisCluster_GetTopology(RedisModuleCtx *ctx) {
     // Handle slots
     ASSERT_KEY(currShard, 0, "slots");
     RedisModuleCallReply *slots = RedisModule_CallReplyArrayElement(currShard, 1);
-    parseSlots(slots, &topo->shards[i]);
+    if (!parseSlots(slots, &topo->shards[topo->numShards])) continue;
 
     // Handle nodes
     ASSERT_KEY(currShard, 2, "nodes");
     RedisModuleCallReply *nodes = RedisModule_CallReplyArrayElement(currShard, 3);
     RS_ASSERT(RedisModule_CallReplyType(nodes) == REDISMODULE_REPLY_ARRAY);
     // parse and store the master
-    parseMasterNode(nodes, &topo->shards[i].node);
+    if (!parseMasterNode(nodes, &topo->shards[topo->numShards].node)) {
+      rm_free(topo->shards[topo->numShards].slotRanges);
+      continue;
+    }
+    // Successfully parsed this shard
+    topo->numShards++;
```

Comment:
Just confirming:
`topo->numShards` may be less than the actual allocated number of shards now, but allocated `shards` will get cleaned up in `MRClusterTopology_Free` just fine.